### ISSUE TITLE
QueueAdvancedTest.testTakeInterruption & testPutInterruption 

### ISSUE THREAD

**Issue by hasancelik** [hazelcast/hazelcast#6297]
```
java.lang.AssertionError: null
	at org.junit.Assert.fail(Assert.java:86)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertTrue(Assert.java:52)
	at com.hazelcast.collection.impl.queue.QueueAdvancedTest.testTakeInterruption(QueueAdvancedTest.java:601)
```

https://hazelcast-l337.ci.cloudbees.com/view/Official%20Builds/job/Hazelcast-3.x-OpenJDK6/672/testReport/

**Comment by Donnerbart** [comment]
I actually have no idea how a `null` value could occur in that line of code. The instance itself is a final `AtomicBoolean` which should not disappear. Also a `AtomicBoolean.get()` in OpenJDK6 returns a `boolean`, not a `Boolean` which could be `null`.

I improved the assert messages to include the instance variable and some text what the `assertTrue` is about via: https://github.com/hazelcast/hazelcast/commit/04b79740eb10ba36e462bf467723b288675bab4a

Also this test may depend on some timing, so it might by sensible to CloudBees performance. Will close until it fails again.

**Comment by hasancelik** [comment]
```
java.lang.AssertionError: Expected queue.take() to be interrupted, but interrupted state is: false
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at com.hazelcast.collection.impl.queue.QueueAdvancedTest.testTakeInterruption(QueueAdvancedTest.java:635)
```

https://hazelcast-l337.ci.cloudbees.com/view/Official%20Builds/job/Hazelcast-3.x-OracleJDK8/com.hazelcast$hazelcast/719/testReport/junit/com.hazelcast.collection.impl.queue/QueueAdvancedTest/testTakeInterruption/

**Comment by hasancelik** [comment]
```
java.lang.AssertionError: Expected queue.put() to be interrupted, but interrupted state is: false
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at com.hazelcast.collection.impl.queue.QueueAdvancedTest.testPutInterruption(QueueAdvancedTest.java:668)
```

https://hazelcast-l337.ci.cloudbees.com/view/Official%20Builds/job/Hazelcast-3.x/com.hazelcast$hazelcast/4856/testReport/junit/com.hazelcast.collection.impl.queue/QueueAdvancedTest/testPutInterruption/

https://hazelcast-l337.ci.cloudbees.com/view/Official%20Builds/job/Hazelcast-3.x/4858/

https://hazelcast-l337.ci.cloudbees.com/view/Official%20Builds/job/Hazelcast-3.x/com.hazelcast$hazelcast/4861/testReport/junit/com.hazelcast.collection.impl.queue/QueueAdvancedTest/testPutInterruption/

**Comment by hasancelik** [comment]
suppressed by https://github.com/hazelcast/hazelcast/commit/415b92c594eae8e44ae3e7099e1b7f89439c3128

**Comment by pveentjer** [comment]
Enabled again in this PR:

When I run the test locally, it works fine. 
https://github.com/hazelcast/hazelcast/pull/8110

**Comment by hasancelik** [comment]
@pveentjer It seems that we have failures again
```
java.lang.AssertionError: error instanceof com.hazelcast.core.OperationTimeoutException: OfferOperation invocation failed to complete due to operation-heartbeat-timeout. Current time: 2016-05-06 13:48:24.879. Total elapsed time: 2250 ms. Last heartbeat: never. Last member heartbeat: 2016-05-06 13:48:24.879. Invocation{op=com.hazelcast.collection.impl.queue.operations.OfferOperation{serviceName='hz:impl:queueService', identityHash=1322919554, partitionId=41, replicaIndex=0, callId=0, invocationTime=1462531702629 (2016-05-06 13:48:22.629), waitTimeout=-1001, callTimeout=1000}, tryCount=250, tryPauseMillis=500, invokeCount=0, callTimeoutMillis=1000, firstInvocationTimeMs=1462531702629, firstInvocationTime='2016-05-06 13:48:22.629', lastHeartbeatMillis=0, lastHeartbeatTime='1970-01-01 02:00:00.000', target=[127.0.0.1]:5570, pendingResponse={VOID}, backupsAcksExpected=0, backupsAcksReceived=0, connection=null}
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at com.hazelcast.test.TestThread.assertFailsEventually(TestThread.java:73)
	at com.hazelcast.collection.impl.queue.QueueAdvancedTest.testPutInterruption(QueueAdvancedTest.java:658)
```

https://hazelcast-l337.ci.cloudbees.com/view/Official%20Builds/job/Hazelcast-3.x/5022/#showFailuresLink

**Comment by gurbuzali** [comment]
From the stack-trace it looks like the failure is caused by an heartbeat problem `OfferOperation invocation failed to complete due to operation-heartbeat-timeout`. 
The logs are lost. I'll close the issue and re-enable the tests so that we can have logs if fails again.

**Comment by gurbuzali** [comment]
looks like the tests are already fixed/enabled via https://github.com/hazelcast/hazelcast/pull/8419

**Comment by Donnerbart** [comment]
Tests are failing again:
```
java.lang.AssertionError: Expected queue.put() to be interrupted, but interrupted state is: false
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at com.hazelcast.collection.impl.queue.QueueAdvancedTest.testPutInterruption(QueueAdvancedTest.java:668)
```
https://hazelcast-l337.ci.cloudbees.com/view/Official%20Builds/job/Hazelcast-3.maintenance-ZuluJDK6/lastCompletedBuild/com.hazelcast$hazelcast/testReport/com.hazelcast.collection.impl.queue/QueueAdvancedTest/testPutInterruption/

```
java.lang.AssertionError: Expected queue.take() to be interrupted, but interrupted state is: false
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at com.hazelcast.collection.impl.queue.QueueAdvancedTest.testTakeInterruption(QueueAdvancedTest.java:635)
```
https://hazelcast-l337.ci.cloudbees.com/view/Official%20Builds/job/Hazelcast-3.maintenance-ZuluJDK6/lastCompletedBuild/com.hazelcast$hazelcast/testReport/com.hazelcast.collection.impl.queue/QueueAdvancedTest/testTakeInterruption/

**Comment by jerrinot** [comment]
@Donnerbart: it appears your failures were in the maintenance branch whilst the [fix](https://github.com/hazelcast/hazelcast/issues/6297) was not backported. 

I'm going to send a backport.

**Comment by jerrinot** [comment]
fixed by #8558
